src/main/java/org/jenkinsci/plugins/displayurlapi/actions/AbstractDisplayAction.java

```diff
@@ -41,13 +41,25 @@ public final Object doRedirect(StaplerRequest req, StaplerResponse rsp) throws I
     protected abstract String getRedirectURL(DisplayURLProvider provider);
 
     DisplayURLProvider lookupProvider() {
-        User user = User.current();
-        DisplayURLProvider provider = (user == null) ? null : user.getProperty(PreferredProviderUserProperty.class).getConfiguredProvider();
+        DisplayURLProvider provider = null;
```

Comment:
Can this be rewritten in a way so that 

```
if (...) { }
else if (..) { } 
else {
  provider = null;
}
return provider;
```

so that we can let the compiler check for initialisation of its variable? safer.